cmake/CPM.cmake

```diff
@@ -31,7 +31,8 @@ cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
 set(CURRENT_CPM_VERSION 1.0.0-development-version)
 
 if(CPM_DIRECTORY)
-  if(NOT CPM_DIRECTORY STREQUAL CMAKE_CURRENT_LIST_DIR)
+  get_filename_component(_CPM_DIRECTORY "${CMAKE_CURRENT_LIST_DIR}" REALPATH)
```

Comment:
- Could we rename the variable something more descriptive like `CPM_CURRENT_DIRECTORY`? 
- Also I think the `get_filename_component` statement can be moved above the `if` statement to avoid duplication, as it exists in both branches.

Reply:
Good points, updated to do both.